diff --git a/minidlna-1.2.1/metadata.c b/minidlna-1.2.1/metadata.c
index 8a10c77..e87887d 100644
--- a/minidlna-1.2.1/metadata.c
+++ b/minidlna-1.2.1/metadata.c
@@ -844,6 +844,10 @@ GetVideoMetadata(const char *path, const char *name)
 			xasprintf(&m.mime, "video/x-matroska");
 		else if( strcmp(ctx->iformat->name, "flv") == 0 )
 			xasprintf(&m.mime, "video/x-flv");
+		else if( strcmp(ctx->iformat->name, "rm") == 0 )
+		    xasprintf(&m.mime, "video/x-pn-realvideo");
+		else if( strcmp(ctx->iformat->name, "rmvb") == 0 )
+		    xasprintf(&m.mime, "video/x-pn-realvideo");
 		if( m.mime )
 			goto video_no_dlna;

diff --git a/minidlna-1.2.1/upnpglobalvars.h b/minidlna-1.2.1/upnpglobalvars.h
index fd2b3db..e92cadb 100644
--- a/minidlna-1.2.1/upnpglobalvars.h
+++ b/minidlna-1.2.1/upnpglobalvars.h
@@ -165,7 +165,8 @@
 	"http-get:*:audio/mp4:*," \
 	"http-get:*:audio/x-wav:*," \
 	"http-get:*:audio/x-flac:*," \
-	"http-get:*:application/ogg:*"
+	"http-get:*:application/ogg:*" \
+	"http-get:*:video/x-pn-realvideo:*"

 #define DLNA_FLAG_DLNA_V1_5      0x00100000
 #define DLNA_FLAG_HTTP_STALLING  0x00200000
diff --git a/minidlna-1.2.1/utils.c b/minidlna-1.2.1/utils.c
index fc21b7f..e00ddac 100644
--- a/minidlna-1.2.1/utils.c
+++ b/minidlna-1.2.1/utils.c
@@ -399,6 +399,7 @@ is_video(const char * file)
 		ends_with(file, ".m2t") || ends_with(file, ".mkv")   ||
 		ends_with(file, ".vob") || ends_with(file, ".ts")    ||
 		ends_with(file, ".flv") || ends_with(file, ".xvid")  ||
+		ends_with(file, ".rm")  || ends_with(file, ".rmvb")  ||
 #ifdef TIVO_SUPPORT
 		ends_with(file, ".TiVo") ||
 #endif
